Framework/LiveData/src/ISIS/FakeISISEventDAE.cpp

```diff
@@ -44,7 +44,8 @@ class TestServerConnection : public Poco::Net::TCPServerConnection {
    * @param soc :: A socket that provides communication with the client.
    */
   TestServerConnection(const Poco::Net::StreamSocket &soc, int nper, int nspec,
-                       int rate, int nevents, std::shared_ptr<Progress> prog)
+                       int rate, int nevents,
+                       const std::shared_ptr<Progress> &prog)
```

Comment:
In this case I would suggest leaving `std::shared_ptr<Progress> prog` as it was and then doing m_prog(std::move(prog)) so we have:

```c++
TestServerConnection(const Poco::Net::StreamSocket &soc, int nper, int nspec,
                     int rate, int nevents, std::shared_ptr<Progress> prog):
  : Poco::Net::TCPServerConnection(soc), m_nPeriods(nper),
    m_nSpectra(nspec), m_Rate(rate), m_nEvents(nevents), m_prog(std::move(prog)) {
```
This helps avoid a copy of the `shared_ptr` on construction of the object in some cases.